spark-wrapper/spark-2.4/src/main/scala/org/apache/spark/sql/TiAggregationImpl.scala

```diff
@@ -30,17 +30,28 @@ object TiAggregationImpl {
     case PhysicalAggregation(groupingExpressions, aggregateExpressions, resultExpressions, child) =>
       // Rewrites all `Average`s into the form of `Divide(Sum / Count)` so that we can push the
       // converted `Sum`s and `Count`s down to TiKV.
-      val (averages, averagesEliminated) =
+      val (averages, _) =
         aggregateExpressions
           .map(_.asInstanceOf[AggregateExpression])
           .partition {
             case AggregateExpression(_: Average, _, _, _) => true
             case _                                        => false
           }
 
+      val (sums, _) = aggregateExpressions.map(_.asInstanceOf[AggregateExpression]).partition {
```

Comment:
this change is to promote sum(long) aggression. This may lead to overflow.

Reply:
same here